src/Application/Services/OpenAiServiceFactory.cs

```diff
@@ -1,45 +1,62 @@
 ﻿using Microsoft.Extensions.Configuration;
 using OpenAI.GPT3;
+using OpenAI.GPT3.Interfaces;
 using OpenAI.GPT3.Managers;
 
 namespace Application.Services;
+
 public class OpenAiServiceFactory
 {
     private readonly IConfiguration _config;
 
+    private IOpenAIService _gpt3Service;
+    private IOpenAIService _gpt4Service;
+
     public OpenAiServiceFactory(IConfiguration config)
     {
         _config = config;
     }
-    public OpenAIService Create(string? azureOpenAiResourceName)
+
+    public IOpenAIService Create(string? azureOpenAiResourceName)
     {
-        var openAiApiKey = _config["OpenAiApiKey"];
-        var azureOpenAiApiKey = _config["AzureOpenAiApiKey"];
-        var azureOpenAiEndpoint = _config["AzureOpenAiEndpoint"];
-        var useAzureOpenAi = _config.GetValue<bool>("UseAzureOpenAiBool");
+        //TODO: Detect whether user is authenticated
```

Comment:
Generally a good practice to leave a link to the PBI for a TODO. Sometimes it's not necessary but in this case as you actually do already have a PBI it couldn't hurt to drop it in here.

Non-blocking comment, just FYI.